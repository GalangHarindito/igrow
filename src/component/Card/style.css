.card {
  display: flex;
  justify-content: center;
}

.card-container {
  background-color: #fff;
  width: 18rem;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  border-radius: 2rem;
  box-shadow: 0px 1rem 1.5rem rgb(0 0 0 / 50%);
  margin-right: 2rem;
  
}

.card-container:last-child {
  margin-right: 0rem;
  
}

.card-image {
  width: 10rem;
  height: 10rem;
  background-repeat: no-repeat;
  background-size: cover;
  border: 0.5rem solid #fff;
  border-radius: 50%;
  -webkit-border-radius: 50%;
  -moz-border-radius: 50%;
  -ms-border-radius: 50%;
  -o-border-radius: 50%;
}

.card-container > div:nth-child(1) {
  background-color: #fff;
  padding: 1rem;
  background-size: cover;
}

.card-container > div:nth-child(1) > div {
  background-color: #fff;
  padding: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  transform: translate(30%, 50%);
  -moz-transform: translate(30%, 50%);
  -ms-transform: translate(30%, 50%);
  -o-transform: translate(30%, 50%);
  -webkit-transform: translate(30%, 50%);
}
.card-container > div:nth-child(2) {
  height: 3.5rem;
}

.card-container > div:nth-child(3) {
  padding: 1.2rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.card-container > div:nth-child(3) > p {
  padding: 0.2rem 0;
}

.name {
  font-weight: 600;
}
.location {
  font-size: 0.8rem;
}

.phone {
  font-size: 0.9rem;
}

@media screen and (max-width: 700px) {
  .card-container {
    margin-right: 0rem;
    margin-bottom: 2rem;
  }

  
}
