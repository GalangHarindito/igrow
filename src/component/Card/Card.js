import React from "react";
import "./style.css";
import background from '../../assets/background.png';
import background2 from '../../assets/cool-background-1.png';

export default function Card(props) {
  const { data } = props;

  return (
    <section className='card'>
      {data.map((el, idx) => {
        return (
          <section className='card-container' key={idx}>
            <div style={{ backgroundImage: el.gender === 'female'? `url(${background2})` : `url(${background})` }}>
              <div
                className='card-image'
                style={{ backgroundImage: `url(${el.picture.large})` }}
              ></div>
            </div>
            <div></div>
            <div>
              <p className='name'>
                {el.name.first} {el.name.last}
              </p>
              <p className='phone'>{el.phone}</p>
              <p className='location'>
                {el.location.city} - {el.location.country}
              </p>
            </div>
          </section>
        );
      })}
    </section>
  );
}
