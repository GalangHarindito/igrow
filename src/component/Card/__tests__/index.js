import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from '../Card';


describe('src/component/Card', () => {
  const data = [
    {
      name : 'test',
      gender: 'test',
      phone: 'test',
      picture: { large:'test' },
      location: { city: 'test', country: 'test'}
    },
    {
      name : 'test',
      gender: 'test',
      phone: 'test',
      picture: { large:'test' },
      location: { city: 'test', country: 'test'}
    }
  ];

  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Card data={data} />);
    expect(tree).toMatchSnapshot();
  });

  test('data', () => {
    const data = [
      {
        name : 'test',
        gender: 'test',
        phone: 'test',
        picture: { large:'test' },
        location: { city: 'test', country: 'test'}
      },
      {
        name : 'test',
        gender: 'test',
        phone: 'test',
        picture: { large:'test' },
        location: { city: 'test', country: 'test'}
      }
    ];
    const result = Card({
      ...Card.defaultProps, data: data
    });
  });

});
