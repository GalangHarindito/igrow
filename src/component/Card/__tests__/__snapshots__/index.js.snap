// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/Card render 1`] = `
<section
  className="card"
>
  <section
    className="card-container"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(background.png)",
        }
      }
    >
      <div
        className="card-image"
        style={
          Object {
            "backgroundImage": "url(test)",
          }
        }
      />
    </div>
    <div />
    <div>
      <p
        className="name"
      >
         
      </p>
      <p
        className="phone"
      >
        test
      </p>
      <p
        className="location"
      >
        test
         - 
        test
      </p>
    </div>
  </section>
  <section
    className="card-container"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(background.png)",
        }
      }
    >
      <div
        className="card-image"
        style={
          Object {
            "backgroundImage": "url(test)",
          }
        }
      />
    </div>
    <div />
    <div>
      <p
        className="name"
      >
         
      </p>
      <p
        className="phone"
      >
        test
      </p>
      <p
        className="location"
      >
        test
         - 
        test
      </p>
    </div>
  </section>
</section>
`;
